subsys/bluetooth/controller/ll_sw/ull_sync.c

```diff
@@ -205,17 +205,17 @@ uint8_t ll_sync_create_cancel(void **rx)
 
 	node_rx = (void *)scan->per_scan.node_rx_estab;
 	link_sync_estab = node_rx->hdr.link;
-	link_sync_lost = sync->node_rx_lost.link;
+	link_sync_lost = sync->node_rx_lost.hdr.link;
 
 	ll_rx_link_release(link_sync_lost);
 	ll_rx_link_release(link_sync_estab);
 	ll_rx_release(node_rx);
 
-	node_rx = (void *)&sync->node_rx_lost;
+	node_rx = (void *)&sync->node_rx_lost.hdr;
```

Comment:
in `ull_conn.c:1716` the node_rx is assigned to the outer struct, rather than the header. 

I think we should consider being consistent with that, so either do the same here, or change the `ull_conn.c` as well.